### ISSUE TITLE
[Feature Request] Opentracing Support

### ISSUE THREAD

**Issue by jmoseley** [temporalio/sdk-node#219]
### Is your feature request related to a problem? Please describe.

We use OpenTracing in our existing worker (written in Go), but have been using the Node client to trigger workflows. Because the client does not yet have Golang support, we do not have full traces for the workflow.

### Describe the solution you'd like

I should be able to leverage my existing opentracing infrastructure to create spans when a workflow is triggered, and have this span encapsulate the entire workflow, including all activities.

**Comment by bergundy** [comment]
This is supported by the SDK but you'll need to implement an opentracing interceptor on your own.
For reference, check out the opentelemetry interceptors package: https://github.com/temporalio/sdk-node/tree/main/packages/interceptors-opentelemetry

**Comment by jmoseley** [comment]
Does the opentelemetry instrumentation integrate well will the existing opentracing instrumentation in the Go SDK? Does the trace IDs cross the boundary? I want to be able to trace between client calls (eg `ExecuteWorkflow`) and see spans in the Workflow with the same trace ID.

**Comment by bergundy** [comment]
Is there an official temporal opentracing library for golang?
The node version uses this header: https://github.com/temporalio/sdk-node/blob/main/packages/interceptors-opentelemetry/src/workflow/index.ts#L19 and expects the value to be an otel `SpanContext`, it's decoded from payload (JSON) using data converter.
https://github.com/open-telemetry/opentelemetry-js-api/blob/a752e7f814e7b335e50705d281cb2c6e1bd38a35/src/trace/span_context.ts#L23

**Comment by jmoseley** [comment]
> Is there an official temporal opentracing library for golang?

Yes, you can pass an Opentracing tracer to the golang SDK client. This gives you traces around activity execution.

https://github.com/temporalio/sdk-go/blob/e8455cbe8014d2587a2b72ce153ae490160d9b11/internal/client.go#L390

**Comment by bergundy** [comment]
I wasn't aware of this, thanks!

IMO the interceptors will probably not be compatible.
We'll have to prioritize adding this, contributions are welcome.

**Comment by Sushisource** [comment]
@jmoseley Where does your opentracing data end up? Opentelemetry exporters most likely exist for wherever that's going to.

As for having the spans connect, even if everything was going through opentracing I'm not sure they'd be attached properly as it stands, that might only work when everything is in Go. That's something I'm going to be looking into pretty soon, at least in the context of getting Opentelemetry spans shared between Node and the underlying rust Core connected.  Getting them connected with server is also on the menu but might take a bit.

**Comment by jmoseley** [comment]
It's flowing to Datadog. They have an opentelemetry solution, but it requires some extra infrastructure that I'd prefer to avoid if possible.

We have spans connecting across languages with API calls originating from Go activities, and I just want to close the loop with API calls that kick off workflows.

**Comment by bergundy** [comment]
The spans connect with opentelemetry if the workflow starter is in **node**.
Activities started from the workflow are also connected to the same parent span.
